File: json_memory_tests/python/flat_10k.py
import json
import time
import os

payload_tbl = []
for i in range(10000):
    payload_tbl.append({"id": i, "name": f"User_{i}", "active": True})

t0 = time.perf_counter()

data = payload_tbl
raw_str = json.dumps(data)
with open('flat_10000_temp.json', 'w') as f:
    f.write(raw_str)

read_str = ""
if os.path.exists('flat_10000_temp.json'):
    with open('flat_10000_temp.json', 'r') as f:
        read_str = f.read()
    os.remove('flat_10000_temp.json')

parsed = json.loads(read_str)
count = len(parsed)
mid_elem = parsed[5000]
name_mid = mid_elem["name"]

t1 = time.perf_counter()
json_ms = (t1 - t0) * 1000.0
print(f"json_elapsed_ms: {json_ms:.3f}")

File: json_memory_tests/ruby/flat_10k.rb
require 'json'
require 'time'

payload_tbl = []
10000.times do |i|
  payload_tbl << { id: i, name: "User_#{i}", active: true }
end

t0 = Time.now

data = payload_tbl
raw_str = JSON.generate(data)
File.write('flat_10000_temp.json', raw_str)

read_str = ""
if File.exist?('flat_10000_temp.json')
  read_str = File.read('flat_10000_temp.json')
  File.delete('flat_10000_temp.json')
end

parsed = JSON.parse(read_str)
count = parsed.length
mid_elem = parsed[5000]
name_mid = mid_elem["name"]

t1 = Time.now
json_ms = (t1 - t0) * 1000.0
puts "json_elapsed_ms: #{'%.3f' % json_ms}"
